### PR TITLE
update modular driver to use BCReader module

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
The BCReader module was implemented by #175. Now the modular driver `coupler_driver_modular.jl` needs to be updated to use this instead of including `coupler_utils/bcfile_reader.jl`.

## To-do
- [x] Replace `include` line with import of BCReader module
- [x] Update calls of functions from BCReader

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Review by LenkaNovak — APPROVED** [review]
Looks good, thanks @juliasloan25 !

**Comment by LenkaNovak** [comment]
Actually, I've just noticed that the Buildkite test is only running this driver for the `slabplanet` mode, so the BC functions are not being called. It would be good to add another Buildkite run of this driver in the `AMIP` mode.

File: experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl [hunk 1]
```diff
@@ -36,7 +36,7 @@ end
 
 BCFileInfo{FT}(args...) where {FT} = BCFileInfo{FT, typeof.(args[1:7])...}(args...)
 
-float_type(::BCFileInfo{FT}) where {FT} = FT
+float_type_bcf(::BCFileInfo{FT}) where {FT} = FT
```
Comment (LenkaNovak) [inline experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl:39]: Nice!

**Review by LenkaNovak — APPROVED** [review]
We now have more stringent criteria for codecov, so the test is failing in `Utilities'jl`. I would delete the `heaviside` function. Looks like we don't need it anymore. Hopefully that'll help it pass.

**Comment by juliasloan25** [comment]
> We now have more stringent criteria for codecov, so the test is failing in `Utilities'jl`. I would delete the `heaviside` function. Looks like we don't need it anymore. Hopefully that'll help it pass.

@LenkaNovak even after removing the `heaviside` function from `Utilities.jl`, codecov still fails. Should I add tests for the functions in `Utilities.jl`, or just merge it as-is?

**Pull request merged**